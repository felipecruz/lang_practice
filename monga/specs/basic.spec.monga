extern void print_int (int _value);

int w;

int[] return_array (int size) {
    int[] array;
    array = new int[size];
    return array;
}

void x () {
    int[] array;
    int a;
    int size;

    size = 10;
    w = -1;
    w = 10;
    {
        int b;
        a = 1;
        b = 1;
        w = 10;
    }

    a = 0;
    array = return_array (size);
    while (a < size) {
        array[a] = size - a;
        a = a + 1;
    }

    a = 0;
    while (a < size) {
        print_int (array[a]);
        a = a + 1;
    }
}

int main () {
    int c;
    int a;
    {
        w = 1;
        a = 1;
        c = 0;
    }

    print_int (w);
    print_int (a);
    print_int (c);

    x ();

    return 1;
}
