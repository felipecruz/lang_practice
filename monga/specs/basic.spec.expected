Checking Func x
Checking assignment types... 
Checking assignment types... 
Checking assignment types... 
Checking Func z
Checking assignment types... 
Get Expression Var Type...
